### PR TITLE
CI: git scrape groups and products

### LINKED ISSUE #18: Git scrape Product and Group information for all agencies
We would like to understand changes over time to the Littlepay Product and Group details for each agency we work with. This has come up recently as we learned about a change to a Product ID after that Product was duplicated to create a new Product -- we didn't expect this to create a TWO new Products 🤷 

So we'd like to capture the output of our `littlepay groups` and `littlepay products` CLI commands, in CSV files stored in this repository, to be able to assess those changes over time more easily.

## Acceptance Criteria

- [ ] For each agency in production (MST, SacRT, SBMTD as of opening this issue)
- [ ] On some schedule (at least once per day)
- [ ] Run `littlepay groups` and `littlepay products`
- [ ] Store the data in a corresponding CSV file in this repository: `data/[agency slug]_[groups|products].csv`
- [ ] With column headers matching the fields in the corresponding [`GroupResponse`](https://github.com/cal-itp/littlepay/blob/main/littlepay/api/groups.py#L9) and [`ProductResponse`](https://github.com/cal-itp/littlepay/blob/main/littlepay/api/products.py#L9) objects
- [ ] Also store the output of `littlepay groups products` in a CSV file in this repository: `data/[agency slug]_linked_groups_products.csv`
- [ ] With column headers `group_id, product_id`
- [ ] A new commit is made each time this process runs, for each agency, when any data file is changed

## Additional context

Sort of related to the work in https://github.com/cal-itp/benefits/pull/1889, can be implemented in workflow in a similar fashion.

More about the [git scraping technique](https://simonwillison.net/2020/Oct/9/git-scraping/).

### PR DESCRIPTION
**Pull request by thekaveman**
Closes #18 

Adds a new flag `--csv` to the `groups` and `products` commands, which formats the output from those commands (i.e. the list of groups/products) into simple CSV rather than the default Python representation.

### CODE REVIEW

File: littlepay/api/groups.py [hunk 1]
```diff
@@ -11,6 +11,16 @@ class GroupResponse:
     label: str
     participant_id: str
 
+    def csv(self) -> str:
+        """Get a CSV str representation of values for this GroupResponse."""
+        return ",".join(vars(self).values())
```
Comment (angela-tran) [inline littlepay/api/groups.py:16]: If one of the values contained `,`, would the CSV be incorrect?

(For a group, I think it'd just be `label` that has the potential to contain `,`)

File: littlepay/api/products.py [hunk 2]
```diff
@@ -14,6 +14,16 @@ class ProductResponse:
     description: str
     participant_id: str
 
+    def csv(self) -> str:
+        """Get a CSV str representation of values for this ProductResponse."""
+        return ",".join(vars(self).values())
```
Comment (angela-tran) [inline littlepay/api/products.py:19]: Same question here about potential `,`s in any of the values for the product

**Review by angela-tran — COMMENTED** [review]
The workflow looks good to me. The implementation and tests for CSV output seem fine to me, except for maybe an edge case that I left questions about. Also just curious, what is the advantage of saving the output as a CSV vs. just the printed strings?

**Comment by thekaveman** [comment]
@angela-tran 

> Also just curious, what is the advantage of saving the output as a CSV vs. just the printed strings?

I thought about this too. I think CSV makes it easier to diff (less noise), and analyze for a non-developer. GitHub will present the CSV file as a table when viewed in GitHub, and it's easy to move into e.g. a spreadsheet. Seemed like a simple enough thing to implement.

Noted on your comments regarding values that may contain commas though!! Thanks for pointing out this edge-case, will add a small workaround.

File: littlepay/api/groups.py (hunk 1 above)
Comment (thekaveman) [reply littlepay/api/groups.py]: Fixed in [b8187d2](https://github.com/cal-itp/littlepay/pull/19/commits/b8187d229627178b882b898f39eaf635eca15488)

File: littlepay/api/products.py (hunk 2 above)
Comment (thekaveman) [reply littlepay/api/products.py]: Fixed in [b8187d2](https://github.com/cal-itp/littlepay/pull/19/commits/b8187d229627178b882b898f39eaf635eca15488)
